trulens_eval/trulens_eval/streamlit.py

```diff
@@ -0,0 +1,255 @@
+                feedback_cols[i].metric(
+                    label=col_name,
+                    value=f"{round(mean, 2)}",
+                    delta=f"{cat.icon} {cat.adjective}",
+                    delta_color=(
+                        "normal" if cat.compare(
+                            mean, CATEGORY.PASS[cat.direction].threshold
+                        ) else "inverse"
+                    ),
+                )
+
+        st.markdown("""---""")
+
+@st.experimental_fragment(run_every=2)
+def trulens_feedback(record: Record):
+    """
+    Render clickable feedback pills for a given record.
+
+    Args:
+
+        record : A trulens record.
+
+    !!! example
+
+        ```python
+        from trulens_eval import streamlit as trulens_st
+
+        with tru_llm as recording:
+            response = llm.invoke(input_text)
+
+        record, response = recording.get()
+
+        trulens_st.trulens_leaderboard()
+        ```
+    """
+    feedback_cols = []
+    feedbacks = {}
+    icons = []
+    for feedback, feedback_result in record.wait_for_feedback_results().items():
+        call_data = {
+            'feedback_definition': feedback,
+            'feedback_name': feedback.name,
+            'result': feedback_result.result
+        }
+        feedback_cols.append(call_data['feedback_name'])
+        feedbacks[call_data['feedback_name']] = FeedbackDisplay(
+            score=call_data['result'],
+            calls=[],
+            icon=_get_icon(fdef=feedback, result=feedback_result.result)
+        )
+        icons.append(feedbacks[call_data['feedback_name']].icon)
+
+    st.write('**Feedback functions**')
+    selected_feedback = pills(
+        "Feedback functions",
+        feedback_cols,
+        index=None,
+        format_func=lambda fcol: f"{fcol} {feedbacks[fcol].score:.4f}",
+        label_visibility=
+        "collapsed",  # Hiding because we can't format the label here.
+        icons=icons,
+        key=
+        f"{call_data['feedback_name']}_{len(feedbacks)}"  # Important! Otherwise streamlit sometimes lazily skips update even with st.experimental_fragment
+    )
+
+    if selected_feedback is not None:
+        st.dataframe(
+            display.get_feedback_result(
+                record, feedback_name=selected_feedback
+            ),
+            use_container_width=True,
+            hide_index=True
+        )
+
+
+def _get_icon(fdef: FeedbackDefinition, result: float) -> str:
+    """
+    Get the icon for a given feedback definition and result.
+
+    Args:
+
+    fdef : FeedbackDefinition
+        The feedback definition
+    result : float
+        The result of the feedback
+
+    Returns:
+        str: The icon for the feedback
+    """
+    cat = CATEGORY.of_score(
+        result or 0,
+        higher_is_better=fdef.higher_is_better
+        if fdef.higher_is_better is not None else True
+    )
+    return cat.icon
+
+def trulens_trace(record_json: dict, app: AppDefinition):
```

Comment:
should this take a Record and transform it here?

Reply:
changed this back to just using `Tru` singleton. Think this is actually okay because of singleton nature